docs/src/model_creation/dsl_advanced.md

```diff
@@ -272,6 +272,39 @@ sol = solve(oprob)
 plot(sol)
 ```
 
+### [Turning off species, parameter, and variable inferring](@id dsl_advanced_options_require_dec)
+In some cases it may be desirable for Catalyst to not infer species and parameters from the DSL, as in the case of reaction networks with very many variables, or as a sanity check that variable names are written correctly. To turn off inferring, simply add the `@require_declaration` macro to one of the lines of the `@reaction_network` declaration. Having this macro means that every single variable, species, or parameter will have to be explicitly declared using the `@variable`, `@species`, or `@parameter` macro. In the case that the DSL parser encounters an undeclared symbolic, it will error with an `UndeclaredSymbolicError` and print the reaction or equation that the undeclared symbolic was found in. 
+
+```@example dsl_advanced_no_infer
+using Catalyst
+# The following case will throw an UndeclaredSymbolicError.
+try @macroexpand @reaction_network begin
+    @require_declaration
+    (k1, k2), A <--> B
+end
+catch e
+    println(e.msg)
+end
+```
+In order to avoid an error in this case all the relevant species and parameters will have to be declared.
+```
+# The following case will not error. 
+t = default_t()
+rn = @reaction_network begin
+    @require_declaration
+    @species A(t) B(t)
+    @parameters k1 k2
+    (k1, k2), A <--> B
+end
+```
+
+The following cases in which the DSL would normally infer variables will all throw errors if `@require_declaration` is set and the variables are not explicitly declared.
+- Inferring a species in a reaction, as in the example above
```

Comment:
I would probably write "occurrence of an undeclared species in a reaction, as in the example above".  As the problem is not about not being able to infer, but rather that stuff is not declared. But maybe I am nitpicking, it is a very slight opinion.